R/html-tweak.R

```diff
@@ -324,6 +324,90 @@ badges_extract_text <- function(x) {
   xml <- xml2::read_html(x)
   badges_extract(xml)
 }
+
+activate_navbar <- function(html, path, pkg) {
+
+  html_navbar <- xml2::xml_find_first(html, ".//div[contains(@class, 'navbar')]")
+  nav_items <- xml2::xml_find_all(html_navbar,".//li[contains(@class, 'nav-item')]")
+
+  # external links can't be an active item
+  keep_internal <- function(links, pkg) {
+    links[!grepl("https?\\:\\/\\/", links) | grepl(pkg$meta$url %||% "", links)]
+  }
+
+  remove_useless_parts <- function(links, pkg) {
+    sub(
+      pkg$meta$url %||% "", "",
+      sub(
+        "^/", "",
+        gsub(
+          "\\.\\.\\/", "",
+          sub(
+            "\\/index.html\\/?", "",
+            links
+          )
+        )
+      )
+    )
+  }
+
+  get_hrefs <- function(nav_item, pkg = pkg) {
+    href <- xml2::xml_attr(xml2::xml_child(nav_item), "href")
+
+    if (href != "#") {
+      links <- href
+    } else {
+      # links in a drop-down
+       hrefs <- xml2::xml_attr(xml2::xml_find_all(nav_item, ".//a"), "href")
+       links <- hrefs[hrefs != "#"]
+    }
+
+   df <- tibble::tibble(
+     nav_item = list(nav_item),
+     links = remove_useless_parts(keep_internal(links, pkg = pkg), pkg = pkg)
+   )
+   row.names(df) <- NULL
+   df
+  }
```

Comment:
I feels like this part could maybe be extract out in to a separate function that just returns the data about the navbar? That would make it a little easier to test, I think.